features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/instrumentation/insights/InsightsCollector.kt

```diff
@@ -110,3 +114,17 @@ interface InsightsUpdatesListener {
      */
     fun onDataUpdated()
 }
+
+/**
+ * Provides the [InsightsCollector] instance inside the SDK, making it accessible to other modules.
+ */
+@InternalApi
+object InsightsCollectorProvider {
+    /**
+     * The nullable singleton [InsightsCollector] instance.
+     */
+    val insightsCollector: InsightsCollector? = (Datadog.getInstance() as? InternalSdkCore)
```

Comment:
Question: This is a val in a singleton. If when first accessed the insightsCollector is null, will it recompute the instance when accessed again?